### ISSUE TITLE
Bodhi no longer able to send fedmsg's correctly

### ISSUE THREAD

**Issue by puiterwijk** [fedora-infra/bodhi#1470]
In commit e9a26042b694e6202dccee8db494b43ce3832244, the FedMsg manager is removed.
While that change in itself is reasonable, the problem is that this new code does not work in general.
It enqueues all kinds of things, but then never actually sends them.

````
[Tue Apr 25 00:14:49.020235 2017] [:error] [pid 21489] 2017-04-25 00:14:49,019 INFO  [bodhi.server][MainThread] fedmsg initialized
[Tue Apr 25 00:14:49.020838 2017] [:error] [pid 21489] 2017-04-25 00:14:49,020 DEBUG [bodhi.server][MainThread] fedmsg enqueueing 'update.edit'
[Tue Apr 25 00:15:13.728408 2017] [:error] [pid 21489] 2017-04-25 00:15:13,728 WARNI [bodhi.server][MainThread] The resource could not be found.
````

**Comment by puiterwijk** [comment]
This is currently worked around in production with http://infrastructure.fedoraproject.org/cgit/ansible.git/commit/?id=9eb2e3d but that is very inefficient, as it makes the busgateway a single point of failure for bodhi messages.
Fedmsg has code to deal with the mod_wsgi case, as long as you make sure that you run fedmsg.init enough in advance.

**Comment by puiterwijk** [comment]
I have currently worked around this on production by updating the `if force:` with `if True or force:`, effectively bypassing the after_commit handler, and then fedmsg sending restored.

**Comment by bowlofeggs** [comment]
It seems that the the ```fedmsg``` key on the ```session.info``` attribute is not present in [```send_fedmsgs_after_commit```](https://github.com/fedora-infra/bodhi/blob/2.6.0/bodhi/server/notifications.py#L54-L79), even though I can see that it is getting set in [```publish```](https://github.com/fedora-infra/bodhi/blob/2.6.0/bodhi/server/notifications.py#L109-L111).

It turns out that we are not getting the same ```Session``` instance during the ```after_commit``` sqlalchemy event as we had during the transaction, so the messages never make it to the ```send_fedmsgs_after_commit``` function.

**Comment by jeremycline** [comment]
Okay, if that's the case, then I think what is happening is someone is publishing fedmsgs which get queued up, but then the session is never committed. If you are seeing a different session committing the expected changes, then the old session is being thrown out for some reason and a new one is being made.

This means either a) the messages shouldn't be tied to a database session because they aren't about changes in the database or b) they _should_ be tied to a session, but currently the session isn't being used correctly.

Do you have a solid way for me to reproduce this locally?

**Comment by bowlofeggs** [comment]
This fix for this has been cherry picked to the 2.6 branch as 1566e99edcaa4ee2246072f1a620f32340560c58.
